src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -385,6 +383,52 @@ public final Builder name(String name) {
 			return this;
 		}
 
+		/**
+		 * Connection pool configuration for a specific remote host.
+		 *
+		 * @param remoteHost the remote host
+		 * @param spec connection pool configuration for a this remote host
+		 * @return {@literal this}
+		 * @throws NullPointerException if remoteHost or/and spec are null
+		 */
+		public final Builder forRemoteHost(SocketAddress remoteHost, Consumer<ConnectionPoolSpec<?>> spec) {
+			Objects.requireNonNull(remoteHost, "remoteHost");
+			Objects.requireNonNull(spec, "spec");
+			ConnectionPoolSpec<?> builder = new ConnectionPoolSpec<>();
+			spec.accept(builder);
+			this.confPerRemoteHost.put(remoteHost, builder);
+			return this;
+		}
+
+		/**
+		 * Builds new ConnectionProvider
+		 *
+		 * @return builds new ConnectionProvider
+		 */
+		public ConnectionProvider build() {
+			return new PooledConnectionProvider(this);
+		}
+	}
+
+	class ConnectionPoolSpec<SPEC extends ConnectionPoolSpec<SPEC>> implements Supplier<SPEC> {
```

Comment:
At first I was not sure the generic self-type helps with code readability here since the only two values of `SPEC` are `Builder` and `?` in the current code. A simple `ConnectionPoolSpec` class that is extended by `Builder` to limit the `name()` `build()` and `forRemoteHost()` methods to the global-level builder could have sufficed, but at the price of duplicating the methods in `ConnectionPoolSpec` to change the return type... So I guess it helps ;)

Maybe add a `forRemoteHost`-specific implementation of `ConnectionPoolSpec`, like `HostSpecificSpec extends ConnectionPoolSpec<HostSpecificSpec>`, and replace `ConnectionPoolSpec<?>` with `HostSpecificSpec` in signatures, for clarity? (and could especially be useful if we later want to add forRemoteHost-specific builder methods)

Reply:
> Maybe add a forRemoteHost-specific implementation of ConnectionPoolSpec, like HostSpecificSpec extends ConnectionPoolSpec<HostSpecificSpec>, and replace ConnectionPoolSpec<?> with HostSpecificSpec in signatures, for clarity? (and could especially be useful if we later want to add forRemoteHost-specific builder methods)

Agree